lib/phoenix/pubsub/pg2_server.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Phoenix.PubSub.PG2Server do
+  use GenServer
+
+  @moduledoc """
+  `Phoenix.PubSub` adapter based on `:pg2`
+
+  See `Phoenix.PubSub.Redis` for details and configuration options.
+  """
+
+  def start_link(opts) do
+    GenServer.start_link __MODULE__, opts, name: Dict.fetch!(opts, :name)
+  end
+
+  def init(opts) do
+    server_name   = Keyword.fetch!(opts, :name)
+    local_name    = Keyword.fetch!(opts, :local_name)
+    pg2_namespace = pg2_namespace(server_name)
+
+    :ok = :pg2.create(pg2_namespace)
+    :ok = :pg2.join(pg2_namespace, self)
+
+    {:ok, %{local_name: local_name, namespace: pg2_namespace}}
+  end
+
+  def handle_call({:subscribe, pid, topic, link}, _from, state) do
+    response = {:perform, {GenServer, :call, [state.local_name, {:subscribe, pid, topic, link}]}}
+    {:reply, response, state}
+  end
+
+  def handle_call({:unsubscribe, pid, topic}, _from, state) do
+    response = {:perform, {GenServer, :call, [state.local_name, {:unsubscribe, pid, topic}]}}
+    {:reply, response, state}
+  end
+
+  def handle_call({:broadcast, from_pid, topic, msg}, _from, state) do
+    case :pg2.get_members(state.namespace) do
```

Comment:
haha. I actually needed to match on `:no_such_group` so we can return `{:error, reason}` to accommodate the pubsub adapter contract. Otherwise, I'd let it crash in horrible ways.